src/main/java/mcp/mobius/opis/modOpis.java

```diff
@@ -93,6 +93,16 @@ public class modOpis {
     public static String commentOpis = "Minimum access level to be open Opis interface. Valid values : NONE, PRIVILEGED, ADMIN";
     public static String commentPrivileged = "List of players with PRIVILEGED access level.";
 
+    public modOpis() throws ReflectiveOperationException {
+        final boolean isHeadless = Boolean.getBoolean("java.awt.headless");
+        if (FMLLaunchHandler.side().isClient() && !isHeadless) {
+            proxy = (ProxyServer) Class.forName("mcp.mobius.opis.proxy.ProxyClient").getConstructor().newInstance();
```

Comment:
Why use a reflective instantiation here? Yes, a simple assignment will fail bytecode verification on Java 8 (because `ProxyClient` will get loaded so `ProxyClient extends ProxyServer` can get verified), but you could also just do multiple casts, I think.
```suggestion
            proxy = (ProxyServer) (Object) new ProxyClient();
```

Do double-check me just in case, I still don't understand how Java 8 class loading works 100%

Reply:
The issue here is that this means java might load the ProxyClient class and run its static initializers etc. when the constructor is executing, and that would cause a crash due to client-only dependencies. Using reflection avoids the somewhat unpredictable class loading and turns it into fully predictable behaviour (it's underspecified in the java spec and hotspot does different things here than some other jvms)

Reply:
@eigenraven Can you put that in a comment?

Reply:
> The issue here is that this means java might load the ProxyClient class and run its static initializers etc. when the constructor is executing, and that would cause a crash due to client-only dependencies. Using reflection avoids the somewhat unpredictable class loading and turns it into fully predictable behaviour (it's underspecified in the java spec and hotspot does different things here than some other jvms)

I would say that about 70% of the mods in GTNH depend on class-loading not triggering until execution reaches the code, although I'm not sure how much of that depends on static initializers vs. the internal code of the methods.